tests/unit/amo/components/TestCollectionAddAddon.js

```diff
@@ -160,16 +180,73 @@ describe(__filename, () => {
     expect(root.find(Notice)).toHaveLength(1);
     expect(root.find(Notice).children()).toHaveText('Added to collection');
 
-    expect(root.instance().props.uiState.addonWasAdded).toEqual(true);
+    expect(root.instance().props.uiState.completedAddonAction).toEqual(
+      addonAddedAction,
+    );
+
+    // Trigger the setTimeout behavior.
+    clock.tick(MESSAGE_RESET_TIME);
+
+    applyUIStateChanges({ root, store });
+    expect(root.instance().props.uiState.completedAddonAction).toEqual(null);
```

Comment:
do we need to test state here too? I guess I was under the impression that checking if it exists (the next line) is enough..?

Reply:
Yes, I suppose what I'm testing here is just the internal implementation, although, because `uiState.completedAddonAction` has three potential values, I kind of like checking that it's been set to what I expect here, in addition to checking that the notice has disappeared.

Reply:
ahh..ok that makes sense